### PR TITLE
Split compile 

### LINKED ISSUE #1083: Two compile methods are redundant
### What is the expected enhancement?
Currently we have:

1. wrapper.compile()
```python
def compile(circuits, backend,
            config=None, basis_gates=None, coupling_map=None, initial_layout=None,
            shots=1024, max_credits=10, seed=None, qobj_id=None, hpc=None,
            skip_transpiler=False)
```

2. transpiler.compile()
```python
def compile(circuits, backend,
            config=None, basis_gates=None, coupling_map=None, initial_layout=None,
            shots=1024, max_credits=10, seed=None, qobj_id=None, hpc=None,
            pass_manager=None)
```

The latter should just be removed. The first will have a `pass_manager` option. 

Also see #978 about other simplifications to these function signatures.

### PR DESCRIPTION
**Pull request by jaygambetta**
### Summary
Fixes #1083

Basically transpiler circuits -> circuits (dags at the moment). The transpile has no qobj and should not need to know about qobj

tools compile circuits -> qobj 

Follow on issues #1144 #1145 and #1146

### CODE REVIEW

File: qiskit/transpiler/_transpiler.py [hunk 1]
```diff
@@ -21,35 +19,28 @@
...
-        max_credits (int): maximum credits to use
-        seed (int): random seed for simulators
-        qobj_id (int): identifier for the generated qobj
+        seed_mapper (int): random seed for the swap_mapper
         hpc (dict): HPC simulator parameters
         pass_manager (PassManager): a pass_manager for the transpiler stage
 
     Returns:
-        QobjExperiment: Experiment to be wrapped in a Qobj.
+        dags: a list of dags.
```
Comment (1ucian0) [inline qiskit/transpiler/_transpiler.py]: Shouldn't `compile` continue returning QobjExperiments?

Comment (jaygambetta) [reply qiskit/transpiler/_transpiler.py]: this is a mistake it should be dag and circuits in the future

**Comment by jaygambetta** [comment]
@ajavadia what do you think about calling the compile in transpile.py transpile and then the function that is a helper in there called transpile we call transpile_dag. I personally think most of these helper functions can be simplified and this is what point four above is about.

**Comment by ajavadia** [comment]
@jaygambetta yes i agree. i like the compile() function to go vertically from circuit to qobj, and the transpile function to go horizontally from circuit to circuit.

This would eventually work very well if we merge DAGCircuit and QuantumCircuit. Then the result of the transpiler can be drawn, you can get the depth, etc. without any further transformations. And you only compile when you want to serialize and send to backend.

For now, I think we can remove `compile.transpile()` and make `transpile()` go circuit to circuit. (A helper for `dag2dag`/`transpile_dag` is fine). But before merging this PR, I think we should make a `_dag2circuit` helper function to achieve this.

**Comment by jaygambetta** [comment]
The only way i could see to make the dag2circuit is the unroller and I’m in two minds.  But I agree making this and circuit2qobjs is a good addition before merging.

**Comment by jaygambetta** [comment]
@ajavadia renamed files.

**Pull request merged**